### PR TITLE
Update to use React Router data router APIs

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Related to https://github.com/tektoncd/dashboard/issues/2452

There will be be at least one additional PR to refactor some of the remaining resource list / resource details containers and move the configuration of the resource GVK to the routes instead of using custom API hooks.

This is all part of a greater effort to reduce boilerplate, which reduces the maintenance overhead for each additional resource type added, and in turn reduces the barrier to entry for contributors as the code is much simplified and easier to modify. It will also provide greater consistency across resources in terms of layout, functionality, etc. Some resources such as *Runs are likely to remain quite different due to the additional content associated with them, but will be able to leverage the simplified config to reduce the amount of custom code required for each of their views.

/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (new features, significant UI changes, API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: src/api/utils.js [hunk 1]
```diff
@@ -235,19 +251,13 @@ export function useSelectedNamespace() {
...
+export function useCollection({ group, kind, params, queryConfig, version }) {
   const { disableWebSocket, ...reactQueryConfig } = queryConfig || {};
-  const query = useQuery(
-    [kind, params].filter(Boolean),
-    () => api(params),
-    reactQueryConfig
-  );
+  const query = useQuery({
+    queryKey: [group, version, kind, params].filter(Boolean),
+    // () => api(params),
```
Comment (briangleeson) [inline src/api/utils.js:258]: Can remove the commented out line now?

File: src/routes/dashboard.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019-2024 The Tekton Authors
```
Comment (briangleeson) [inline src/routes/dashboard.jsx:2]: New file only needs 2024, like the other new files?
```suggestion
Copyright 2024 The Tekton Authors
```

**Review by briangleeson — CHANGES_REQUESTED** [review]
Thanks for the walkthrough. Couple of nits but otherwise looks good overall

File: src/routes/dashboard.jsx (hunk 2 above)
Comment (AlanGreene) [reply src/routes/dashboard.jsx:2]: The content is largely copied from the old route definitions in App.jsx which is why I kept the original date, just restructured to follow the new API

Comment (AlanGreene) [reply src/routes/dashboard.jsx:2]: although I realise now that I didn't do the same for the pipelines and triggers routes… 🙈 
I'll update them all for consistency. Thanks

File: src/api/utils.js (hunk 1 above)
Comment (AlanGreene) [reply src/api/utils.js]: Done

**Review by briangleeson — APPROVED** [review]
/lgtm

**Pull request merged**